server/s3_delete_object_action.cc

```diff
@@ -273,15 +333,42 @@ void S3DeleteObjectAction::delete_object() {
     next();
     return;
   }
-  // process to delete object
-  if (!motr_writer) {
-    motr_writer = motr_writer_factory->create_motr_writer(
-        request, object_metadata->get_oid());
+  unsigned int object_count = del_object_oids.size();
+  if (object_count > 0) {
+    // process to delete object
+    std::pair<struct m0_uint128, int>& obj_layout_pair = del_object_oids.back();
+    if (!motr_writer) {
+      motr_writer = motr_writer_factory->create_motr_writer(
+          request, obj_layout_pair.first);
+    } else {
```

Comment:
is this else needed ?

Reply:
yes, else part is required to set correct oid to be deleted. Now, we have responsibility to delete multiple objects.

Reply:
But if motr_writer is not set then create_motr_writer() instantiate it after that setting of OID needed right